Bluetooth: ISO: Expose CIG_ID, CIS_ID, BIG_Handle and BIS_Number

Thalley opened the pull request:
Expose the IDs, handles and numbers of CIGs, BIGs, CISes and BISes
via the ISO info struct. This information may be needed by higher
layer (such as the Basic Audio Profile (BAP)) or used as indexes
for arrays in the application layer.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/94380

Thalley committed:
Bluetooth: ISO: Change bis_id to bis_number

Modify the bis_id in struct bt_conn_iso to be bis_number
and correctly initialize it for both the broadcaster and
the sync receiver.

The field is currently unused and the term bis_id does not
exist in the core spec. The BIS Index was renamed to BIS
Number in Core 6.1, and thus the Zephyr code base will
reflect that.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Expose CIG_ID, CIS_ID, BIG_Handle and BIS_Number

Expose the IDs, handles and numbers of CIGs, BIGs, CISes and BISes
via the ISO info struct. This information may be needed by higher
layer (such as the Basic Audio Profile (BAP)) or used as indexes
for arrays in the application layer.

The location of the CIG_ID, CIS_ID, BIG_Handle and BIS_Number
are moved to the location struct inside of struct bt_conn_iso
instead of being stored directly in it.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
tests: Bluetooth: ISO: Verify ranges of BIG handles and BIS Numbers

Verify that the BIG handles and BIS Numbers in the ISO info
struct are valid.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
doc: releases: Add entry for new ISO info fields

Add entry for new fields added to the ISO info structs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: BAP: Replace bt_bap_iso with bt_iso_chan for streams

Modify the struct bt_bap_stream to have a direct reference to the
bt_iso_chan, rather than having a reference to the parent
bt_bap_iso.

This change will allow users to more easily and directly access
the bt_iso_chan, instead of getting the reference via
bt_bap_ep_get_info.
Additionally, this allows users to access the bt_iso_chan for
streams that are not codec configured, and thus not have a
bt_bap_ep reference.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
doc: releases: Add entry for new iso field in bt_bap_stream

Add entry for the new field added to the bt_bap_stream.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
samples: Bluetooth: BAP: Unicast: Print CIG and CIS IDs

Update the unicast client and server samples to use
bt_iso_chan_get_info to get the CIG and CIS IDs of a
BAP stream.

The samples show how it is possible to get the information early
as the unicast client (it is possible right after the stream
has been added to a group), and how the values are only
available to the server once the CIS is connected, which
is indicated by the stream_started callback.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -3074,12 +3074,19 @@ int bt_iso_big_create(struct bt_le_ext_adv *padv, struct bt_iso_big_create_param
+		/* BIS numbers start from 1, so the first BIS will get BIS Number = 1 */
+		const uint8_t bis_number = i + 1U;
+
+		err = big_init_bis(big, param->bis_channels[i], bis_number, true);
```
my fault... this  do not need to change... only once dereferenced in the loop.

Thalley commented:
The PR needs additional work for the BAP section, as `stream->iso` isn't set by the unicast server, broadcast sink or broadcast source

Thalley committed:
samples: Bluetooth: BAP: Broadcast: Print BIG and BIS IDs

Update the broadcast sink and source samples to use
bt_iso_chan_get_info to get the BIG handle and BIS number of a
BAP stream.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

larsgk reviewed (dismissed):
LGTM - but didn't try it out
